internal/candidate_runner/runner.go

```diff
@@ -175,18 +180,20 @@ func (runner *candidateRunner) Run(info *groups.RunnerInfo) error {
 
 			candidatesName = append(candidatesName, node.Name)
 
-			if !runner.dryRun {
-				logForNode.Info("Adding drain candidate taint")
-				if _, errTaint := k8sclient.AddNLATaint(ctx, runner.client, node, runner.clock.Now(), k8sclient.TaintDrainCandidate); errTaint != nil {
-					logForNode.Error(errTaint, "Failed to taint node")
-					continue // let's try next node, maybe this one has a problem
+			if dataInfo.CircuitBreakersOk {
```

Comment:
It feels wrong to now couple observability data with the actual logic. Right now, dataInfo is only used to provide context to the cli and it doesn't matter if it works or not.

Reply:
It is precisely because I consider circuitBreak status as part of the context that I have added it dataInfo, this way when we use the CLI to get group information we know if it was circuit-breaked or not.
I would not qualify this as "Observability", it is an active layer that changes the behavior of the runner.

Reply:
But it wasn't used like this so far 🤔 
It was only gathering information about the recent run so that we are able to provide it to the cli later on. I get the idea of having a single source of truth, but we could archive the same by:
```suggestion
			if runner.areCircuitBreakerOk() {
			  dataInfo.CircuitBreakersOk = true
```

[Here](https://github.com/DataDog/draino/blob/2704b62a33ad377ec753514f0ddf807a24316aac/internal/candidate_runner/runner.go#L153-L154) we have a similar case

Reply:
There is a reason why I don't want to call this `areCircuitBreakerOk()` function in the for loop:
It is not a readonly func, in case of half-open state it will consume a token on the rateLimiter, and I want this token to be consumed for the group not for a single node!
This is why I need a bool to store once I compute it out of the loop... and since I want also to propagate this information in the CLI for group, I am using the dataInfo.

I will add a comment about that.

Reply:
Ahh that makes a lot of sense. Thanks for the explanation!

> I will add a comment about that.

That would be awesome